babbage/cube.py

```diff
@@ -131,5 +145,37 @@ def compute_cardinalities(self):
             result = self.members(dimension.ref, page_size=0)
             dimension.spec['cardinality'] = result.get('total_member_count')
 
+    def restrict_joins(self, q, bindings):
+        """
+        Restrict the joins across all tables referenced in the database
+        query to those specified in the model for the relevant dimensions.
+        If a single table is used for the query, no unnecessary joins are
+        performed. If more than one table are referenced, this ensures
+        their returned rows are connected via the fact table.
+        """
+        if len(q.froms) == 1:
+            return q
+        else:
+            for binding in bindings:
+                if binding.table == self.fact_table:
+                    continue
+                concept = self.model[binding.ref]
+                if isinstance(concept, Dimension):
+                    dimension = concept
+                else:
+                    dimension = concept.dimension
```

Comment:
I guess that at some point we might want that, especially if we add more operators to cuts.
e.g. 'find all rows where measure > 1000000'

However with current cuts it's really not useful. When we get to that feature we'll handle this as well.